engines/router/missionctl/server/upi/logutils_test.go

```diff
@@ -52,3 +59,94 @@ func Test_logTuringRouterRequestSummary(t *testing.T) {
 		})
 	}
 }
+
+// TestLogTuringRouterRequestSummary tests that the response and request are parsed into the expected format
+// this include nested proto where underscore separated is expected e.g. prediction_result_table
+func TestLogTuringRouterRequestSummary(t *testing.T) {
+	table := &upiv1.Table{
+		Name: "abc",
+		Rows: []*upiv1.Row{
+			{
+				RowId: "row1",
+				Values: []*upiv1.Value{
+					{
+						DoubleValue: 123.456,
+					},
+				},
+			},
+		},
+	}
+	header := metadata.New(map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	})
+	headerString := "map[key1:value1 key2:value2]"
+
+	tests := []struct {
+		name              string
+		reqHeader         metadata.MD
+		reqBody           *upiv1.PredictValuesRequest
+		resHeader         metadata.MD
+		resBody           *upiv1.PredictValuesResponse
+		expectedReqBody   string
+		expectedReqHeader string
+		expectedResHeader string
+		expectedResBody   string
+	}{
+		{
+			name:      "ok",
+			reqHeader: header,
+			reqBody: &upiv1.PredictValuesRequest{
+				PredictionTable: table,
+			},
+			resHeader: header,
```

Comment:
Would be better if the `resHeader` and `reqHeader` are different values, to also test that they are not wrongly captured in the logs, no?